src/FileDownloader/ChunkFileDownloader.cs

```diff
@@ -80,31 +87,58 @@ public async Task DownloadAsync(CancellationToken ct = default)
             try
             {
                 _downloadSpeedStopwatch.Start();
-
-                OnProgressChanged();
-
                 DownloadSize = await GetDownloadSizeAsync();
 
                 var readRanges = CalculateFileChunkRanges();
+                var fileRanges = readRanges as FileRange[] ?? readRanges.ToArray();
+                _chunkDownloadQueue = new ConcurrentQueue<FileRange>(fileRanges);
 
-                //Parallel download
-                using (new Timer(ReportProgress, null, 100, 100))
-                {
-                    await OrchestrateDownloadWorkersAsync(readRanges, ct);
-                    _downloadSpeedStopwatch.Stop();
+                var tasks = Enumerable.Range(1, Math.Min(_concurrentDownloads, _chunkDownloadQueue.Count)).Select(
+                    async workerIndex =>
+                    {
+                        if (workerIndex > 1)
+                            await Task.Delay(1000 * (workerIndex - 1), ct);
+
+                        await DequeueAndDownloadChunksAsync(ct);
+                    });
```

Comment:
I think that the code is much harder to read and understand now because of several reasons:
1. By merging `OrchestrateDownloadWorkersAsync` we are mixing several conceptual elements into this function, the scope is now much larger.
2. The casing of `fileRanges` is a very technical detail that adds noise when reading the code

I much rather keep this as a separate function to make the code much easier to read and follow